analyses/telomerase-activity-prediction/01-run-EXTEND.R

```diff
@@ -32,6 +37,35 @@ output_file <- opt$output
 
 
 PTBA_GE_Standard = readRDS(input_file)
+
+# Filter data if tumor purity threshold is turned on
+if (opt$apply_tumor_purity_threshold) {
+
+  # tumor purity metadata file which has been filtered to only biospecimens that
+  #  survive the cancer-group-level threshold
+  tumor_purity_file <- file.path(
+    rprojroot::find_root(rprojroot::has_dir(".git")),
+    "analyses",
+    "tumor-purity-exploration",
+    "results",
+    "thresholded_rna_stranded_same-extraction.tsv"
+  )
+
+  # Define vector of IDs to retain
+  bs_ids <- readr::read_tsv(tumor_purity_file) %>%
+    dplyr::pull(Kids_First_Biospecimen_ID) %>%
+    unique()
+
+  # Subset the `PTBA_GE_Standard` variable to those IDs
+  PTBA_GE_Standard <- PTBA_GE_Standard[,bs_ids]
+
+  # Quick check on filtering:
+  if (!length(bs_ids) == ncol(PTBA_GE_Standard)) {
+    stop("Error filtering IDs to use in EXTEND calculation with tumor purity thresholding on.")
+  }
```

Comment:
I'm not sure what this check on length is accomplishing for us.

If `PTBA_GE_Standard` is a data frame (hard to tell given the context in the PR alone), would we expect an informative-ish error if an identifier in `bs_ids` was not present in the column names? 

Would it be better to check if the column names are identical to the vector of identifiers?

Reply:
Yep, this was a CI-fixing adventure I had yesterday in modules where this was a df (and not a matrix like here)! This check should just look for names then, but checks where it's a df in other PRs' thresholding notebooks aren't needed.

As I write this, I'm now wondering if we could do with some more modularity. I'd like your thoughts @jaclyn-taroni for this -  I have some form of this code (identical except for whether the subsetting is on a matrix or df, which is easy to handle!) floating around all the thresholding notebooks I've been writing. I'm now thinking this should be a `util` function in the tumor purity module instead that each of those notebooks can call if filtering is turned on. I would make that change here, and then we'd probably just wait to merge this until tackling the other ones.

Reply:
A `util` function in the tumor purity module module makes sense to me. But I'd recommend breaking that out into it's own PR intended to go in first, then propagate the use of that function in the currently open PRs. It gives us more flexibility in getting things in, which seems good.

Reply:
I'm going to slap a don't merge label on the open PRs...